ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -824,12 +824,12 @@ isHiFileStableRule = defineEarlyCutoff $ RuleNoDiagnostics $ \IsHiFileStable f -
 
 displayTHWarning :: LspT c IO ()
 displayTHWarning
-  | isMac && not hostIsDynamic = do
+  | not isWindows && not hostIsDynamic = do
       LSP.sendNotification SWindowShowMessage $
         ShowMessageParams MtInfo $ T.unwords
-          [ "This HLS binary does not support Template Haskell."
+          [ "This HLS binary might not support Template Haskell."
           , "Follow the [instructions](" <> templateHaskellInstructions <> ")"
-          , "to build an HLS binary with support for Template Haskell."
+          , "to build an HLS binary with reliable support for Template Haskell."
```

Comment:
I'm not sure about the wording changes, it seems to me that it will generate more confusion than it's worth. 

Just looking at the last couple of days in the issue tracker to see how recurrent the problem is:

* https://github.com/haskell/haskell-language-server/issues/2596
* https://github.com/haskell/haskell-language-server/issues/2054
* https://github.com/haskell/haskell-language-server/issues/2340
* https://github.com/haskell/haskell-language-server/issues/1497
* https://github.com/haskell/haskell-language-server/issues/2633

Reply:
Well, the fact it *works* for some cases.
Not sure in how many cases is working cause happy users dont open issues but maybe more than we expected.
We are gonna lie to those users and make them install from source when it would not be necessary.

Maybe we can rewrite it to be precise, recommending strongly building from source without ignoring it could work in some environments

Reply:
Ok i changed the first verb from `might`to `could` to denote the binary *likely* will not work with th:
```
"This HLS binary could not support Template Haskell."
          , "Follow this [instructions](" <> templateHaskellInstructions <> ")"
          , "to build an HLS binary with reliable support for Template Haskell."
```

Reply:
I still think that `could` is still confusing, to be honest. 

Taking a step back, your main issue here is that you don't want users to build HLS from source unnecessarily. Why is that so bad? One obvious problem is that `cabal install` will place the binary under `~/.cabal/bin` and will not annotate it with the ghc version number, which means that the vscode extension will not find it. 

So the problem is that instructing users to build from source leads to a bad experience. Should we focus on addressing that, rather than on this wording?

Reply:
Since statically linking glibc is unsupported, I don't know a good way of making the warning more precise.

Reply:
I am afraid my knowledge of those low level details is very limited so i cant help much.
If we cant be precise with the warning i still think we should be cautious with the message.
Users with a working th installation watching that message will lose a little bit of confidence in our error handling.

But well, as you still think the original wording is better will restore it.